### PR TITLE
Add test, update dto and schema, change some endpoint for categories.

### PR DESCRIPTION
**Pull request by namirinz**
Change endpoint.
- Remove `updateRank` because we don't need it `use patch` endpoint instead
- Add `pushMenuToCategory` function to add menu into categories `menus` field

Update DTO and schema information.
Add cucumber test.

### CODE REVIEW

File: src/categories/dto/category.updateCategory.dto.ts [hunk 1]
```diff
@@ -1,18 +1,26 @@
...
   @ApiProperty({ type: String, isArray: true })
   @Transform(({ value }) => value.map((v) => new Types.ObjectId(v)))
   menus: Types.ObjectId[];
 
-  @ApiProperty()
-  @IsNumber()
+  @ApiProperty({
+    type: Number,
+    description: 'Category rank',
+    default: null,
```
Comment (ishiasaka) [inline src/categories/dto/category.updateCategory.dto.ts]: Add field required: false to tell that is a optional field

Comment (namirinz) [reply src/categories/dto/category.updateCategory.dto.ts]: Done

File: src/schema/categories.schema.ts [hunk 2]
```diff
@@ -1,22 +1,26 @@
...
-  @ApiProperty()
+  @ApiProperty({ type: String, description: 'Category title', required: true })
   title: string;
 
-  @Prop({ default: [] })
-  @ApiProperty({ type: String, isArray: true })
-  menus: Types.ObjectId[];
+  @Prop({ default: [], ref: () => MenuSchema })
+  @ApiProperty({ type: String, description: 'List of Menu', isArray: true })
+  menus: Ref<MenuSchema, string>[];
```
Comment (Npwskp) [inline src/schema/categories.schema.ts]: this should be `Ref<MenuSchema>` ?

**Review by Npwskp — COMMENTED** [review]
seems bad

**Comment by namirinz** [comment]
Mergeable @Npwskp

**Review by JiMeow — APPROVED** [review]
Temporary backend

**Pull request merged**